### DISCUSSION TITLE
An Inconvenience

### DISCUSSION THREAD

**Discussion by hkolbeck** [kdl-org/kdl#370] [General]
Hi folks,

I know I haven't been active here in a bit. I was really disappointed that I just wasn't able to focus well enough to maintain https://github.com/kdl-org/kdl4j or https://github.com/kdl-org/kdl-gen. I found out a few weeks ago that the cause of that lack of focus was a large growth in the left temporal lobe of my brain. There are many unknowns, but in all likelihood I will not be alive in a few months. 

I've been really fond of the shape of KDL since I first saw it. It rounded so many sharp corners on the other document languages I'd used. It was a truly wonderful experience to participate in the early conversations about specifics and also the general shape of the language. To watch a group of people who'd mostly never met approach complex questions as without a right answer, only a best-possible answer, healed so many memories of being in the room with engineers screaming at each other. I am a trans woman, and I felt welcome here, and I thank you all for that.

I hope folks will take on maintenance for those repos, that would be a kindness. I have to say I think kdl-gen could be quite a useful tool for folks developing KDL libraries. If they no longer serve, I hope you'll put them to rest and build having learned from my mistakes.

Love,
Hannah 🏳️‍⚧️

**Comment by zkat** [comment]
Hannah 💚

I can't tell you how invaluable your participation and support in the development of KDL as a language has been. Not just the implementations and tools you've built, but just the fact that an entire conformance suite has come from your efforts and has helped the growing list of implementations agree on what KDL is supposed to be.

I'm heartbroken by these news, and stubborn in that I want to hope for the best as long as I can. You're a kind, friendly individual and a talented engineer and I couldn't have asked for a better partner in crime for this little project we all embarked on.

As KDL heads into 2.0, we'll want to update all the existing implementations to comply, and that's going to mean kdl4j as well--and I'll make sure to look for someone who can pick up that work and help it over the finish line. Java is a significant enough language that I'm sure we'll find someone eventually.

Please let me know if there's anything else you need, and if you ever want to talk more about stuff, you're free to reach out to me directly, too (easiest is probably masto or my gh email, but lmk if something else works better for you).

Thank you again. It warms my heart that I could help make this project a warm memory for you.

Take care, and good luck,
kat 💚
